internal/services/iotcentral/iotcentral_application_resource.go

```diff
@@ -134,7 +134,7 @@ func resourceIotCentralAppCreate(d *pluginsdk.ResourceData, meta interface{}) er
 
 	displayName := d.Get("display_name").(string)
 	if displayName == "" {
-		displayName = id.ResourceGroupName
+		displayName = id.ResourceName
```

Comment:
Given we are changing the existing default behaviour can we make this a 4.0 change and wrap it in the flag?